File: types/fsharp/Math.fs
module Math

let factorial numberToCalculateFactorial =
    let mutable accumulator = numberToCalculateFactorial

    for number = numberToCalculateFactorial - 1 downto 1 do
        accumulator <- accumulator * number
    accumulator

let rec factorialWithPatternMatching number =
    match number with
    | 0 | 1 -> 1
    | 2 -> 2
    | _ -> number * factorialWithPatternMatching (number - 1)

File: types/csharp/Math.cs
namespace csharp
{
    class Math
    {
        public static int Factorial(int numberToCalculateFactorial)
        {
            int accumulator = numberToCalculateFactorial;
            
            for(int number = numberToCalculateFactorial - 1; number >= 1; number--)
            {
                accumulator = accumulator * number;
            }

            return accumulator;
        }

        public static int RecursiveFactorial(int number)
        {
            return number == 0 || number == 1 ? 1
                : number == 2 ? 2
                : number * RecursiveFactorial(number - 1);
        }
    }
}